src/OrchardCore/Orchard.ContentManagement/DefaultContentManager.cs

```diff
@@ -303,13 +297,19 @@ public Task<ContentItem> GetAsync(string contentItemId)
             if (latestVersion != null)
             {
                 latestVersion.Latest = false;
-                buildingContentItem.Number = latestVersion.Number + 1;
```

Comment:
Why isn't Latest the higher version number?

Reply:
When you `DiscardDraftAsync()` a draft which is the latest and the highest version, the current published item becomes the latest but without being the highest version